### PR TITLE
Anthropic: Support tools

### PR DESCRIPTION
**Pull request by ghost**
## Context
Anthropic recently began supporting tools: https://docs.anthropic.com/claude/docs/tool-use

## Change
This PR adds support for tools in `AnthropicChatModel`.
`AnthropicStreamingChatModel` does not support tools yet, because Anthropic does not support tools when streaming.

## Checklist
Before submitting this PR, please check the following points:
- [X] Test in AiServicesWithToolsIT
- [X] I have added unit and integration tests for my change
- [X] All unit and integration tests in the module I have added/changed are green
- [X] All unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules are green
- [X] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml) (only when a new module is added)

## Checklist for adding new embedding store integration
- [ ] I have added a {NameOfIntegration}EmbeddingStoreIT that extends from either EmbeddingStoreIT or EmbeddingStoreWithFilteringIT

### CODE REVIEW

File: docs/docs/tutorials/6-tools.md [hunk 1]
```diff
@@ -121,29 +121,82 @@ Please note that tools/function calling is not the same as [JSON mode](/tutorial
...
+- `ToolSpecifications.toolSpecificationsFrom(Object)`
+- `ToolSpecifications.toolSpecificationFrom(Method)`
+
+```java
+class WeatherTools { 
+  
+    @Tool("Returns the weather forecast for a given city")
+    String getWeather(
+            @P("The city for which the weather forecast should be returned") String city,
+            TemperatureUnit temperatureUnit
```
Comment (daixi98) [inline docs/docs/tutorials/6-tools.md:162]: We can parse Enum, but nested object parameters are not supported yet. Is it correct?

**Review by daixi98 — APPROVED** [review]
LGTM

File: docs/docs/tutorials/6-tools.md (hunk 1 above)
Comment (langchain4j) [reply docs/docs/tutorials/6-tools.md:162]: Yes, enums are supported: https://github.com/langchain4j/langchain4j/blob/1827302342eee393fddfce8a07f3cab2b7d5b9e2/langchain4j-core/src/test/java/dev/langchain4j/agent/tool/ToolSpecificationsTest.java#L60

Nested POJOs will also be supported soon: https://github.com/langchain4j/langchain4j/pull/708

File: langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicMapper.java [hunk 2]
```diff
@@ -44,52 +114,29 @@ static String toAnthropicSystemPrompt(List<ChatMessage> messages) {
...
                 .map(content -> content.text)
                 .collect(joining("\n"));
-        return AiMessage.from(text);
+
+        List<ToolExecutionRequest> toolExecutionRequests = contents.stream()
+                .filter(content -> "tool_use".equals(content.type))
+                .map(content -> ToolExecutionRequest.builder()
+                        .id(content.id)
+                        .name(content.name)
+                        .arguments(GSON.toJson(content.input))
```
Comment (geoand) [inline langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicMapper.java:129]: This unfortunately breaks the cases where GSON is not used (like Quarkus which swaps the JSON handling for Jackson in the supported modules).

Comment (langchain4j) [reply langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicMapper.java:129]: Oh right! Let me fix this

Comment (geoand) [reply langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicMapper.java:129]: 🙏🏼

Comment (langchain4j) [reply langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicMapper.java:129]: Using Gson [here](https://github.com/langchain4j/langchain4j/blob/d118c0c3b67e01bb2f159cf7a388533dac06ffad/langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/DefaultAnthropicClient.java#L168) is ok for you?

Comment (geoand) [reply langchain4j-anthropic/src/main/java/dev/langchain4j/model/anthropic/AnthropicMapper.java:129]: That one never gets called in the Quarkus case (or at least I didn't see it), so it should be fine

**Pull request merged**